addons/delivery_mondialrelay/views/views.xml

```diff
@@ -0,0 +1,30 @@
+<odoo>
+  <data>
+
+    <record id="view_delivery_carrier_form_with_provider_mondialrelay" model="ir.ui.view">
+        <field name="name">delivery.carrier.form.provider.mondialrelay</field>
+        <field name="model">delivery.carrier</field>
+        <field name="inherit_id" ref="delivery.view_delivery_carrier_form"/>
+        <field name="arch" type="xml">
+            <field name="product_id" position="after">
+                <field name="is_mondialrelay" invisible="1" />
+                <field name="mondialrelay_brand" attrs="{'required': [('is_mondialrelay', '=', True)], 'invisible': [('is_mondialrelay', '!=', True)]}" />
+            </field>
+        </field>
+    </record>
+
+    <record id="view_partner_form_mondialrelay" model="ir.ui.view">
+        <field name="name">view_partner_form_mondialrelay</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='contact_addresses']//kanban/field[@name='display_name']" position="after">
+                <field name="is_mondialrelay"/>
+            </xpath>
+            <xpath expr="//t[@t-set='delivery_icon_path']" position="after">
+                <t t-set="delivery_icon_path" t-if="record.is_mondialrelay.raw_value" t-value="'/delivery_mondialrelay/static/src/img/truck_mr.png'"/>
```

Comment:
We can also probably do it for the regular kanban view (which doesnt work right now as this is broken since 14.4
=> fix incoming at https://github.com/odoo/odoo/pull/75540
I guess if mondialrelay is merged before my fix, we can add that after in another PR :shrug: 

![aabba](https://user-images.githubusercontent.com/30048408/130794879-9a40b195-d14c-4065-8752-61d9db372474.png)

Reply:
will be fixed after 75540 ;)